coderd/database/migrations/000059_app_sharing_level.up.sql

```diff
@@ -0,0 +1,15 @@
+-- Add enum app_sharing_level
+CREATE TYPE app_sharing_level AS ENUM (
```

Comment:
what about `app_access_level`

Reply:
The terraform key is `share` so I don't think we should use a different word otherwise it'll be harder to grep for it.

Reply:
This is certainly pedantry territory I just find `sharing` to be the wrong intent for what this is. Sharing is a side effect of what we're doing with this work, what this is actually doing is dictating _access_ to the workspace e.g. `sharing_level_owner` is clearly not sharing.

Reply:
Kyle and I discussed this in DMs a week ago and I said:
> if I was just reading "application access" [in the docs] I wouldn't know what it meant
> if I read "application sharing" I'd be like "ok this feature lets you share applications with other users"`

And he agreed that it should probably be called sharing for discovery. Thoughts @kylecarbs @ammario @sharkymark

Reply:
I do agree that "sharing" doesn't apply to the "owner" level, but I think it's fine for it to be called sharing even if it sometimes isn't.